### PR TITLE
add private drishti_gltest lib for cross platform opengl context

### PR DESCRIPTION
**Pull request by headupinclouds**
NOTE: iOS and Android still need a non QT solution

* Add drishti_gltest for portable windowless GLContext
  + GLFWContext (NOT (IOS OR ANDROID))
  + QGLContext (NOT (IOS OR ANDROID)), but should most likely work
* global query replace in filenames, C++, and CMakeLists.txt: ’s/qtplus/gltest/g’
* disable qt + mobile (IOS OR ANDROID) GPU gtests for drishti_{hci,acf,drishti}
* control clang formatting w/ on/off switches
* disable logging during gtests
  + add —silent option to turn loggin goff
  + add option to name logger for FaceTracker
  + remove unused threshold shader param for Nms2

### CODE REVIEW

File: src/lib/drishti/CMakeLists.txt [hunk 1]
```diff
@@ -214,84 +221,59 @@ else()
 
 endif(DRISHTI_BUILD_WORLD)
 
+# We use the internal drishti_gltest for an off screen OpenGL context.  This may be
+# implemented with Qt (portably) or GLFW (NOT (ANDROID OR IOS)).  Qt requires
+# some build overhead and may not be suitable for CI testing.  For Android platforms,
+# Qt will actually build each native "application" as a shared library, which currently
+# doesn't play well with the drishti_test_main module used for cross platform testing,
+# so we skip Android testing for now and can rely on Emulator/QEMU testing.
+if((DRISHTI_BUILD_QT AND NOT (ANDROID OR IOS)) OR (NOT ${drishti_is_mobile}))
```
Comment (ruslo) [inline src/lib/drishti/CMakeLists.txt]: I think it's the same as `if(NOT drishti_is_mobile)`

File: src/lib/drishti/gltest/CMakeLists.txt [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+endif()
+
+if(${drishti_has_glfw})
+  target_compile_definitions(drishti_gltest PUBLIC DRISHTI_HAS_GLFW=1)
+endif()
+
+target_include_directories(drishti_gltest
+  PUBLIC
+  "$<BUILD_INTERFACE:${DRISHTI_INCLUDE_DIRECTORIES}>"
+  "$<INSTALL_INTERFACE:${drishti_include_install_dir}>")
```
Comment (ruslo) [inline src/lib/drishti/gltest/CMakeLists.txt]: We are not installing this library so `INSTALL_INTERFACE` part not needed.

File: src/lib/drishti/CMakeLists.txt (hunk 1 above)
Comment (headupinclouds) [inline src/lib/drishti/CMakeLists.txt]: Yep.  Missed that one.  Thanks.

File: src/lib/drishti/gltest/CMakeLists.txt (hunk 2 above)
Comment (headupinclouds) [inline src/lib/drishti/gltest/CMakeLists.txt]: Will remove it.

**Comment by headupinclouds** [comment]
I just configured a Trusty VM w/ gcc 5.0 and ran a clean build with `gcc-5-pic-hid-sections-lto`:

```
Log saved: /home/dhirvonen/devel/drishti/_logs/polly/gcc-5-pic-hid-sections-lto/log.txt
-
Generate: 0:03:15.371589s
Build: 0:04:38.652168s
-
Total: 0:07:54.024251s
-
SUCCESS
Finished with status 0
```

The travis gcc build is reporting errors like this:

```
    CMake Error at src/app/facecrop/CMakeLists.txt:3 (add_executable):

      Target "drishti-facecrop" links to target "OpenGL::GL" but the target was

      not found.  Perhaps a find_package() call is missing for an IMPORTED

      target, or an ALIAS target is missing?
```

I did encounter OpenGL issues initially in the VM, which were resolved by walking through this setup:

https://nextdime.wordpress.com/2014/06/29/setting-up-an-opengl-mesa-development-environment-ubuntu-14-04/

I also encountered an OpenCV Jasper issue, which I disabled in the OpenCV args for `*nix` platforms for now, since it isn't critical.

I should probably try to reproduce the Travis image exactly.

**Comment by ruslo** [comment]
You can use Docker:
* https://github.com/hunter-packages/travis-linux-docker

Additionally you have to install extra packages manually:
* https://github.com/elucideye/drishti/blob/f33276db37576d315a4f0f073e0396db9216e142/.travis.yml#L16-L32

You can do it by (inside Docker):
```
> sudo add-apt-repository ppa:ubuntu-toolchain-r/test
> sudo apt-get -y update # may fail with unrelated errors
> sudo apt-get -y install g++-5 gcc-5
```

**Comment by ruslo** [comment]
> You can do it by (inside Docker)

Additionally you have to install:
```
> sudo apt-get -y install libglu1-mesa-dev
```

will add this one to image soon.

**Comment by ruslo** [comment]
> will add this one to image soon

Updated: https://github.com/hunter-packages/travis-linux-docker/commit/a8990bf58bab75be6227d86830726184f3767cfa

You [can do](https://github.com/hunter-packages/travis-linux-docker#update):
```
> docker pull quay.io/ruslo/hunter-travis-trusty
```

> Additionally you have to install extra packages manually

Added to README:
* https://github.com/hunter-packages/travis-linux-docker#extra-software

**Comment by headupinclouds** [comment]
Thanks.  I'll start from that image.

**Pull request closed without merging**